### PR TITLE
Allow to build `Consumer` and `Producer` from Java instances

### PR DESCRIPTION
**Pull request by guizmaii**
Based on https://github.com/conduktor/zio-kafka/pull/73

### CODE REVIEW

File: src/main/scala/zio/kafka/consumer/RebalanceConsumer.scala [hunk 1]
```diff
@@ -1,10 +1,9 @@
 package zio.kafka.consumer
 
+import org.apache.kafka.clients.consumer.{ Consumer => Jconsumer, OffsetAndMetadata }
```
Comment (sderosiaux) [inline src/main/scala/zio/kafka/consumer/RebalanceConsumer.scala]:
```suggestion
import org.apache.kafka.clients.consumer.{ Consumer => JConsumer, OffsetAndMetadata }
```

**Comment by sderosiaux** [comment]
general question: will we be able to continue updating from upstream if we change too many things in our fork? (they keep adding improvements & fixes, so we should port them back?)

**Comment by guizmaii** [comment]
@sderosiaux 

We're backporting almost everything we're changing. This will be backported too. 
If you look at zio-kafka, the CDK devs has 5 PRs open there ;)

File: src/main/scala/zio/kafka/consumer/internal/ConsumerAccess.scala [hunk 2]
```diff
@@ -31,21 +28,21 @@ private[consumer] class ConsumerAccess(
...
       .fork
       .flatMap(fib => fib.join.onInterrupt(ZIO.effectTotal(consumer.wakeup()) *> fib.interrupt))
+
 }
 
 private[consumer] object ConsumerAccess {
-  type ByteArrayKafkaConsumer = KafkaConsumer[Array[Byte], Array[Byte]]
+  type ByteArrayKafkaConsumer = JConsumer[Array[Byte], Array[Byte]]
 
-  def make(settings: ConsumerSettings): RManaged[Blocking, ConsumerAccess] =
```
Comment (strokyl) [inline src/main/scala/zio/kafka/consumer/internal/ConsumerAccess.scala:39]: :red_circle: 
So there isn't the old `make` anymore?

Comment (guizmaii) [reply src/main/scala/zio/kafka/consumer/internal/ConsumerAccess.scala:39]: It's for internal use only. This shouldn't be used by zio-kafka users

**Review by gurinderu — APPROVED** [review]

**Pull request merged**